src/PS5Parser.h

```diff
@@ -0,0 +1,392 @@
+
+ Contact information
+ -------------------
+
+ Kristian Lauszus, TKJ Electronics
+ Web      :  http://www.tkjelectronics.com
+ e-mail   :  kristianl@tkjelectronics.com
+
+ PS5 support by Joseph Duchesne
+ */
+
+#ifndef _ps5parser_h_
+#define _ps5parser_h_
+
+#include "Usb.h"
+#include "controllerEnums.h"
+#include "PS5Trigger.h"
+
+/** Buttons on the controller */
+const uint8_t PS5_BUTTONS[] PROGMEM = {
+        UP, // UP
+        RIGHT, // RIGHT
+        DOWN, // DOWN
+        LEFT, // LEFT
+
+        0x0C, // SHARE
+        0x0D, // OPTIONS
+        0x0E, // L3
+        0x0F, // R3
+
+        0x0A, // L2
+        0x0B, // R2
+        0x08, // L1
+        0x09, // R1
+
+        0x07, // TRIANGLE
+        0x06, // CIRCLE
+        0x05, // CROSS
+        0x04, // SQUARE
+
+        0x10, // PS
+        0x11, // TOUCHPAD
+};
+
+union PS5Buttons {
+        struct {
+                uint8_t dpad : 4;
+                uint8_t square : 1;
+                uint8_t cross : 1;
+                uint8_t circle : 1;
+                uint8_t triangle : 1;
+
+                uint8_t l1 : 1;
+                uint8_t r1 : 1;
+                uint8_t l2 : 1;
+                uint8_t r2 : 1;
+                uint8_t share : 1;
+                uint8_t menu : 1;
+                uint8_t left_stick : 1;
+                uint8_t right_stick : 1;
+                
+
+                uint8_t ps : 1;
+                uint8_t touchpad : 1;
+                uint8_t mic : 1;
+                uint8_t dummy : 5;
+        } __attribute__((packed));
+        uint32_t val : 24;
+} __attribute__((packed));
+
+struct touchpadXY {
+        struct {
+                uint8_t counter : 7; // Increments every time a finger is touching the touchpad
+                uint8_t touching : 1; // The top bit is cleared if the finger is touching the touchpad
+                uint16_t x : 12;
+                uint16_t y : 12;
+        } __attribute__((packed)) finger[2]; // 0 = first finger, 1 = second finger
+} __attribute__((packed));
+
+struct PS5Status {
+        // first byte
+        uint8_t headphone : 1;
+        uint8_t dummy : 2;
+        uint8_t usb : 1;  // charging
+        uint8_t dummy2: 4;
+
+        // second byte
+        uint8_t battery : 4;
+        uint8_t dummy3 : 1;
+        uint8_t battery_full : 1;
+        uint8_t dummy4 : 2;
+} __attribute__((packed));
+
+struct PS5Data {
+        /* Button and joystick values */
+        uint8_t hatValue[4];  // 0-3 bytes
+        uint8_t trigger[2];  // 4-5
+        
+        uint8_t dummy;  // 6 unknown
+        
+        PS5Buttons btn;      // 7-9
+
+        uint8_t dummy2[5];   // 0xA-0xD unknown
+
+        /* Gyro and accelerometer values */
+        int16_t gyroY, gyroZ, gyroX;   // 0x0F - 0x14
+        int16_t accX, accZ, accY;      // 0x15-0x1A
+
+        uint8_t dummy3[5];  // 0x1B - 0x1F unknown
+
+        // 0x20 - 0x23 touchpad point 1
+        // 0x24 - 0x27 touchpad point 2
+        touchpadXY xy[2];
```

Comment:
Should be:

```C++
touchpadXY xy;
```

As the two fingers are already defined in the touchpadXY struct.